scripts/configure-venue-access.ts

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env node -r esm -r ts-node/register
+
+import fs from "fs";
+import admin from "firebase-admin";
+import "firebase/firestore";
+
+import {
+  VenueAccessCodes,
+  VenueAccessEmails,
+  VenueAccessMode,
+} from "../src/types/VenueAcccess";
+
+import { initFirebaseAdminApp } from "./lib/helpers";
+
+const mergeStringArrays = (
+  arr1: string[] = [],
+  arr2: string[] = []
+): string[] => {
+  const result = arr1.map((val) => val.trim());
+  arr2.forEach((val) => {
+    const valTrimmed = val.trim();
+    if (!result.includes(valTrimmed)) {
+      result.push(valTrimmed);
+    }
+  });
+  return result;
+};
+
+function usage() {
+  console.log(`
+${process.argv[1]}: Configure venue access. Supports configuring a secret password, list of emails which can access the venue, or ticket codes.
+
+Usage: npx ts-node ${process.argv[1]} PROJECT_ID VENUE_ID [password|emaillist|codelist] [password | emails file path | codes file path]
+
+Example: npx ts-node ${process.argv[1]} co-reality-map password abc123
+Example: npx ts-node ${process.argv[1]} co-reality-map emails emails-one-per-line.txt
+Example: npx ts-node ${process.argv[1]} co-reality-map codes ticket-codes-one-per-line.txt
+`);
+  process.exit(1);
+}
+
+const argv = process.argv.slice(2);
+if (argv.length < 4) {
+  usage();
+}
+
+const projectId = argv[0];
+const venueId = argv[1];
+const method = argv[2];
+const accessDetail = argv[3];
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+if (!(Object as any).values(VenueAccessMode).includes(method)) {
+  console.error(`Invalid access method ${method}`);
+  process.exit(1);
+}
+
+initFirebaseAdminApp(projectId)(async () => {
+  console.log(`Ensuring ${venueId} access via ${method} - ${accessDetail}`);
+  const venueDoc = await admin.firestore().doc(`venues/${venueId}`).get();
+  if (!venueDoc.exists) {
+    console.error(`venue ${venueId} does not exist`);
+    process.exit(1);
+  }
+
+  await admin
+    .firestore()
+    .doc(`venues/${venueId}`)
+    .update({ access: method ?? "" });
```

Comment:
Firebase doesn't allow `undefined` which leaves us with `null` or empty string.

Reply:
I'm not a huge fan of changing data types but if the current convention is to use null as default value will do.

Reply:
I’d prefer it to be ‘unset’ in firestore if it’s a default value, rather than null (I basically don’t ever want to use null if I can help it, as it doesn’t map nicely into TypeScript types/etc)

Reply:
@denisdimitrov Did we have a resolution on this? Or just leaving it as an empty string is fine?